packages/gallery/src/components/gallery/proGallery/galleryHelpers.js

```diff
@@ -1,5 +1,22 @@
 import { window } from 'pro-gallery-lib';
 
+// Promise-wrapping class
+export class Deferred {
+  constructor() {
+    this.promise = new Promise((resolve, reject) => {
+      this.reject = () => {
```

Comment:
hey, there should be a fix here. this.reject(...args) =>{  reject(...args)} same with resolve
export class Deferred {
  constructor() {
    this.promise = new Promise((resolve, reject) => {
      this.reject = (...args) => {
        this.isPending = false;
        reject(...args);
      };
      this.resolve = (...args) => {
        this.isPending = false;
        resolve(...args);
      };
      this.isPending = true;
    });
  }
}